File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/CredentialsApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.model.VerifiableCredential
import my.ssdid.sdk.domain.vault.Vault

class CredentialsApi internal constructor(private val vault: Vault) {
    suspend fun store(credential: VerifiableCredential): Result<Unit> = vault.storeCredential(credential)
    suspend fun list(): List<VerifiableCredential> = vault.listCredentials()
    suspend fun getForDid(did: String): List<VerifiableCredential> = vault.getCredentialsForDid(did)
    suspend fun delete(credentialId: String): Result<Unit> = vault.deleteCredential(credentialId)
}

File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/CredentialsApi.swift
import Foundation

/// Facade for verifiable credential storage operations.
public struct CredentialsApi {
    private let vault: Vault

    init(vault: Vault) {
        self.vault = vault
    }

    public func store(_ credential: VerifiableCredential) async throws {
        try await vault.storeCredential(credential)
    }

    public func list() async -> [VerifiableCredential] {
        await vault.listCredentials()
    }

    public func getForDid(_ did: String) async -> [VerifiableCredential] {
        await vault.getCredentialsForDid(did)
    }

    public func delete(credentialId: String) async throws {
        try await vault.deleteCredential(credentialId: credentialId)
    }
}
